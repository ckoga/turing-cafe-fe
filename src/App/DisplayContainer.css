.res-display {
  display: flex;
  flex-wrap: wrap;
}