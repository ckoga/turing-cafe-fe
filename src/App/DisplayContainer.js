import React from 'react';
import './DisplayContainer.css';
import ReservationCard from './ReservationCard';
import PropTypes from 'prop-types';

const DisplayContainer = ({ data }) => {
  const card = data.map(res => {
    return <ReservationCard 
            key={res.id}
            name={res.name}
            date={res.date}
            time={res.time}
            numberGuests={res.number}
    />
  })

  return (
    <main className='res-display'>
      {card}
    </main>
  )
}

export default DisplayContainer;

DisplayContainer.propTypes = {
  data: PropTypes.array
}