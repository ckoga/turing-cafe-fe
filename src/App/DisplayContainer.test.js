import React from 'react';
import DisplayContainer from './DisplayContainer';
import { shallow } from 'enzyme';

describe('DisplayContainer', () => {
  let wrapper, mockState;
  beforeEach(() => {
    mockState = {
      reservations: [
        {
          name: 'Thao',
          date: '02/09',
          time: '10:16',
          number: 2
        }
      ]
    }
    wrapper = shallow(<DisplayContainer data={mockState.reservations}/> )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  
})