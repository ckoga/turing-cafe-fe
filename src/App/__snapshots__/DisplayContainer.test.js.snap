// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayContainer should match the snapshot 1`] = `
<main
  className="res-display"
>
  <ReservationCard
    date="02/09"
    name="Thao"
    numberGuests={2}
    time="10:16"
  />
</main>
`;
